packages/patternfly-4/react-core/src/components/Nav/NavGroup.js

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import styles from '@patternfly/patternfly-next/components/Nav/styles.css';
+import a11yStyles from '@patternfly/patternfly-next/utilities/Accessibility/styles.css';
+import { css } from '@patternfly/react-styles';
+import PropTypes from 'prop-types';
+import NavToggle from './NavToggle';
+import { AngleRightIcon } from '@patternfly/react-icons';
+
+const propTypes = {
+  id: PropTypes.string.isRequired,
+  title: PropTypes.string.isRequired,
+  ariaLabel: PropTypes.string.isRequired,
+  expandable: PropTypes.bool.isRequired,
+  expanded: PropTypes.bool.isRequired,
+  children: PropTypes.node.isRequired,
+  innerClassName: PropTypes.string.isRequired,
+  groupId: PropTypes.number.isRequired,
+  active: PropTypes.bool.isRequired
+};
+
+const NavGroup = ({
+  id,
+  title,
+  ariaLabel,
+  expandable,
+  expanded: defaultExpanded,
+  children,
+  innerClassName,
+  groupId,
+  active,
+  ...props
+}) => {
+  if (expandable) {
+    return (
+      <NavToggle defaultValue={defaultExpanded}>
+        {({ value: isExpanded, toggle }) => (
+          <li
+            className={css(
+              styles.navItem,
+              isExpanded && styles.modifiers.expanded,
+              active && styles.modifiers.current,
+              innerClassName
+            )}
+            onClick={toggle}
+            {...props}
+          >
+            <a
+              className={css(styles.navLink)}
+              id={ariaLabel ? null : id}
```

Comment:
can you just set this to id

Reply:
ariaLabel is used in the `<h2>` below. And this id attribute should only be added to the DOM if ariaLabel is not set. If ariaLabel is not set, aria-labelledby gets the value from this `<a>` tag. If ariaLabel is set, aria-labelledby gets the value from the `<h2>` tag.

Reply:
I think it is the naming that is throwing me off.  Maybe rename it to sectionTitle since that is what it really is. I think we always need the aria labeled by and therefore always need the id.  I am basing my assumptions off what in see in core.  Maybe I missed some changes in core along the way.
@jgiardino what are you thoughts?

Reply:
the h2 section is not visible and only there for screen readers. The `<section>` always has aria-labelledby with an id either from the `<a>` tag or from the `<h2>`.

Reply:
Also, this is always set since the id is generated so the connection for screen readers will always be there.

Reply:
Sounds good, I can rename it to srText